src/st/strategy/shortcode/class-wpml-pb-shortcode-content-wrapper.php

```diff
@@ -250,4 +250,65 @@ private function insert_wrappers() {
 			$offset          = $offset + mb_strlen( $wrapper );
 		}
 	}
+
+	/**
+	 * @param string $content
+	 * @param array  $shortcodes
+	 *
+	 * @return string
+	 */
+	public static function maybeWrap( $content, array $shortcodes ) {
+		return self::shouldBeWrapped( $content, $shortcodes ) ? self::wrap( $content ) : $content;
```

Comment:
@strategio You could also inline the functions and then use a maybe pipeline.

```php

$containsOneShortcode = pipe( Str::match( '/' . get_shortcode_regex( $shortcodes ) . '/s' ), Fns::isEmpty(), Fns::not() ); 

$withoutShortcodesIsDifferent = curryN( 2, function( $shortcodes, $content ) {
    ....
    return (bool) ....;
} ) );

return Maybe::of( $content )
   ->filter( $containsOneShortcode( $shortcodes) )
   ->filter( $withoutShortcodesIsDifferent )
   ->map( [ self::class 'wrap' ] )
   ->getOrElse( $content );
```

Reply:
Thanks for the tip @brucepearson :)

I implemented this but now I need a way to download the composer dependencies coming from GitLab because the FP repo is private. I will do this as soon as Andrea gives me a token.